hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -239,8 +239,9 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
     //file reader for reading a hudi base file that needs to be merged with log files
     val preMergeBaseFileReader = if (isMOR) {
       // Add support for reading files using inline file system.
-      super.buildReaderWithPartitionValues(sparkSession, dataSchema, partitionSchema,
-        requiredSchemaWithMandatory, requiredFilters, options, new Configuration(hadoopConf))
+      super.buildReaderWithPartitionValues(sparkSession, dataSchema, partitionSchema, requiredSchemaWithMandatory,
+        if (shouldUseRecordPosition) requiredFilters else filters ++ requiredFilters,
```

Comment:
We should further check for presence of record key fields in `filters` and if present, then only add record key fields. For other fields, there maybe some correctness issue.

Reply:
Also, if you have verified this is working, please add the same check and push down filters in `NewHoodieParquetFileFormat` as well.

Reply:
Sounds good.